### PR TITLE
Videos2

### PR DESCRIPTION
**Pull request by keyvanamiri**

### CODE REVIEW

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/MainFragment.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ private ListRow createCardRow(CardRow cardRow) {
 
     private void setupUIElements() {
         setTitle(getString(R.string.browse_title));
-        setBadgeDrawable(getResources().getDrawable(R.drawable.title_android_tv, null));
+        setBadgeDrawable(getResources().getDrawable(R.drawable.title_android_tv));
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/MainFragment.java:90]: Resources.getDrawable(int) is deprecated. You should prefer Context.getDrawable() or passing a valid theme to the 2-arg version.

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 2]
```diff
@@ -49,50 +47,51 @@
...
     private final Context mContext;
+    protected FragmentActivity mActivity;
     private final PlaybackControlsRow.RepeatAction mRepeatAction;
     private final PlaybackControlsRow.ShuffleAction mShuffleAction;
     protected PlaybackControlsRow mControlsRow;
     private static final int REFRESH_PROGRESS = 1;
+    private int mSpeed = PLAYBACK_SPEED_PAUSED;
+    private long mLastTimeUpdate = 0;
+    private int mLastPosition = 0;
+    private static final int[] sFastForwardSpeeds = {5, 10, 20};
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:64]: nit: static final is usually ALL_CAPS

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 3]
```diff
@@ -49,50 +47,51 @@
...
 
     protected MediaMetaData mMediaMetaData = null;
 
-    public MediaPlayerGlue(Context context, PlaybackOverlayFragment fragment) {
-        super(context, fragment, new int[]{1});
+    public MediaPlayerGlue(Context context, PlaybackOverlaySupportFragment fragment,
+                           FragmentActivity activity) {
+        super(context, fragment, sFastForwardSpeeds);
         mContext = context;
+        mActivity = activity;
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:87]: Is this expected to differ from mContext?

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 4]
```diff
@@ -182,31 +181,41 @@ private void queueNextRefresh() {
...
-        consume = consume && System
-                .currentTimeMillis() - mLastKeyDownEvent > FAST_FORWARD_REWIND_REPEAT_DELAY;
-        if (consume) {
-            mLastKeyDownEvent = System.currentTimeMillis();
-            int newPosition = getCurrentPosition() + FAST_FORWARD_REWIND_STEP;
-            if (mSelectedAction instanceof PlaybackControlsRow.RewindAction) {
-                newPosition = getCurrentPosition() - FAST_FORWARD_REWIND_STEP;
+    @Override
+    public boolean onKey(View v, int keyCode, KeyEvent event) {
+        return super.onKey(v, keyCode, event);
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:186]: If you don't do anything here you don't need this method.

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 5]
```diff
@@ -182,31 +181,41 @@ private void queueNextRefresh() {
...
+
+    public int updateCurrentPosition() {
+        long currentTimeMillis = System.currentTimeMillis();
+        int newPosition = getCurrentPosition();
+        if (mSpeed >= PLAYBACK_SPEED_FAST_L0 || mSpeed <= -PLAYBACK_SPEED_FAST_L0) {
+            // only update the current position if we are in fast-forward or rewind mode
+            int speed = 0;
+            if (mSpeed >= PLAYBACK_SPEED_FAST_L0) {
+                speed = getFastForwardSpeeds()[mSpeed - PLAYBACK_SPEED_FAST_L0];
+            } else if (mSpeed <= -PLAYBACK_SPEED_FAST_L0) {
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:197]: This condition will be always true.

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 6]
```diff
@@ -247,6 +254,25 @@ private void queueNextRefresh() {
         // Not supported.
     }
 
+    /**
+     * Callback called after the current media item finishes playing.
+     */
+    void onMediaItemComplete() {
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:260]: Is this a call from an interface? Missing @Override?

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 7]
```diff
@@ -247,6 +254,25 @@ private void queueNextRefresh() {
...
+        // the current song finishes playing.
+        mSpeed = PLAYBACK_SPEED_NORMAL;
+        mLastPosition = 0;
+        mLastTimeUpdate = System.currentTimeMillis();
+    }
+
+    /**
+     * Callback called after the entire medialist finishes playing.
+     */
+    void onMediaListComplete() {
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:271]: Same question as above

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 8]
```diff
@@ -290,28 +329,16 @@ public void setMediaMetaData(MediaMetaData mediaMetaData) {
...
-        } else {
-            mSelectedAction = null;
+    @Override
+    public int getUpdatePeriod() {
+        int totalTime = getControlsRow().getTotalTime();
+        Fragment fragment = getFragment();
+        if (fragment.getView() == null || fragment.getView().getWidth() == 0 || totalTime <= 0) {
+            return 1000;
         }
+        return Math.max(16, totalTime / fragment.getView().getWidth());
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:339]: Meaningful constants for these magic numbers would be nice.

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicConsumptionExampleFragment.java [hunk 9]
```diff
@@ -287,6 +291,14 @@ public MusicConsumptionExampleFragment() {
         }
     }
 
+    /**
+     * This is a listener implementation for the {@link OnItemViewSelectedListener} of the {@link
+     * PlaybackOverlaySupportFragment}. This implementation is required in order to detect KEY_DOWN events
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicConsumptionExampleFragment.java:296]: Are newlines allowed to split @link?

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicConsumptionExampleFragment.java [hunk 10]
```diff
@@ -287,6 +291,14 @@ public MusicConsumptionExampleFragment() {
...
+     * on the {@link android.support.v17.leanback.widget.PlaybackControlsRow.FastForwardAction} and
+     * {@link android.support.v17.leanback.widget.PlaybackControlsRow.RewindAction}.
+     * @see OnItemViewSelectedListener#onItemSelected(Presenter.ViewHolder, Object,
+     * RowPresenter.ViewHolder, Object)
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicConsumptionExampleFragment.java:300]: Are newlines allowed to split @see?

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/settings/SettingsExampleFragment.java [hunk 11]
```diff
@@ -32,7 +32,7 @@
 
 public class SettingsExampleFragment extends LeanbackSettingsFragment implements DialogPreference.TargetFragment {
 
-    private final Stack<Fragment> fragments = new Stack<Fragment>();
+    protected final Stack<Fragment> fragments = new Stack<Fragment>();
```
Comment (nicoya) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/settings/SettingsExampleFragment.java:35]: It's not clear how these changes are related to the description of the CL

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/MainFragment.java (hunk 1 above)
Comment (Karthikg206) [reply app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/MainFragment.java:90]: In Sample Music Player forward and rewind button not working how to solve this?

**Comment by Karthikg206** [comment]
In Sample Music Player forward and rewind button not working how to solve this?
mSelectedAction always return null. how to fix? anyone hlp

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java (hunk 4 above)
Comment (Karthikg206) [reply app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java:186]: In Sample Music Player forward and rewind button not working how to solve this?
mSelectedAction always return null. how to fix? anyone hlp

**Comment by codingjeremy** [comment]
This sample has been migrated to a new location and updates are happening there (check README for more information).

[As recommended by GitHub][1], we are closing all issues and pull requests now that this older repo will be archived. 

If you still see this issue in the updated repo, please reopen the issue/PR there. Thank you!

[1]: https://help.github.com/en/articles/archiving-repositories

**Pull request closed without merging**